### PR TITLE
Bump IOGX v2.0

### PR DESCRIPTION
**Pull request by zeme-wana**
Notable changes:
- Removed `stylish-haskell`,`hindent` and `tidyrc` in favour of `fourmolu` and `editorconfig`.
- Added support for adding extra flags to `cabal-fmt` (namely `--no-tabular`)
- Upgraded IOGX to new interface https://github.com/input-output-hk/iogx/pull/2
- Now building `oci-images` in hydra

### CODE REVIEW

**Comment by zeme-wana** [comment]
In order to start using fourmolu and editorconfig, run `pre-commit run --all` after checking out this PR and entering `nix develop`.
Modify the `.editorconfig` and `fourmolu.yaml` files  accordingly.
`hlint` is also enabled and will return tens of warnings -> edit `.hlint.yaml` accordingly.
It's possible that `fourmolu` and `editorconfig` will fight one another -> you can disable editorconfig in `./nix/pre-commit-check.nix`

File: nix/hydra-jobs.nix [hunk 1]
```diff
@@ -0,0 +1,24 @@
+{ inputs, inputs', pkgs, flake }:
+
+{
+  excludedPaths = [
+    "packages.ghc8107.pre-commit-check"
+    "packages.ghc8107-mingwW64.pre-commit-check"
```
Comment (zeme-wana) [inline nix/hydra-jobs.nix:6]: Temporarily disable pre-commit-check in CI.
A new PR is needed where fourmolu and possibly editorconfig are run against the entire codebase. After that, we can bring back pre-commit-check in CI

Comment (jhbertra) [reply nix/hydra-jobs.nix:6]: Can you add this as a `FIXME comment in the file?

File: nix/iogx-config.nix [hunk 2]
```diff
@@ -0,0 +1,15 @@
+{
+  debug = true;
```
Comment (jhbertra) [inline nix/iogx-config.nix:2]: Should this be `false`?

File: nix/hydra-jobs.nix [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+
+    "devShells.ghc8107-profiled"
+    "devShells.ghc8107-mingwW64-profiled"
+
+    "checks.ghc8107-profiled"
+    "checks.ghc8107-mingwW64-profiled"
+
+    "networks"
+    "nomadTasks"
+    "operables"
```
[the remark this answers is not in the document]
Comment (zeme-wana) [reply nix/hydra-jobs.nix:19]: Ok let's see what hydra thinks

File: nix/iogx-config.nix (hunk 2 above)
Comment (zeme-wana) [reply nix/iogx-config.nix:2]: no difference, it's disappearing in the next push to this PR.

**Comment by zeme-wana** [comment]
I removed `networks` and `nomadTasks` from CI because they are not derivations and so they can't be built.

However `oci-images` and `operables` are now part of CI.

Note that `nix/marlowe-cardano/networks.nix` has problems:
```
  testnet-pioneers = mkNetwork environments.marlowe-pioneers // {
    magic = 1567;
  };
  testnet-dev = mkNetwork environments.marlowe-dev // {
    magic = 1566;
  };
```
`environments.marlowe-pioneers` and `environments.marlowe-dev` are invalid: `marlowe-pioneers` and `marlowe-dev` are missing from `environments` from the more recent version of iohk-nix.
See here for the list of available environments:
https://github.com/input-output-hk/iohk-nix/blob/02f42375ee5c2bab1640f14c6389b7e91bbfec8b/cardano-lib/default.nix#L85

Some `oci-images` derivations also fail to build.
Test this locally:  `nix build .#hydraJobs.oci-images.all.copyToDockerDaemon.x86_64-linux`

File: fourmolu.yaml [hunk 4]
```diff
@@ -0,0 +1,17 @@
...
+newlines-between-decls: 1
+haddock-style: single-line
+haddock-style-module: null
+let-style: inline
+in-style: right-align
+single-constraint-parens: always
+unicode: never
+respectful: false
+fixities: []
+reexports: []
```
[the remark this answers is not in the document]
Comment (jhbertra) [reply fourmolu.yaml:17]: This change seems to have been reverted

**Pull request merged**